packages/devtools_app/lib/src/shared/framework/screen_controllers.dart

```diff
@@ -39,11 +44,15 @@ class ScreenControllers {
   }) {
     final lazyController = _LazyController<T>(creator: controllerCreator);
     if (offline) {
-      assert(!_offlineControllers.containsKey(T));
-      _offlineControllers[T] = lazyController;
+      if (offlineControllers.containsKey(T)) {
+        offlineControllers.remove(T)?.dispose();
+      }
+      offlineControllers[T] = lazyController;
```

Comment:
[optional] consider pulling this logic (and below logic for `controllers`) into shared method